### PR TITLE
build(aio): output `{@example}` tags as `<code-example>` elements

### PR DESCRIPTION
**Pull request by petebacondarwin**

### CODE REVIEW

**Comment by petebacondarwin** [comment]
There is a related PR in the migrator that brings over the linenum formatting correctly: https://github.com/petebacondarwin/aio-migrator/pull/5

PTAL @wardbell

File: aio/transforms/examples-package/inline-tag-defs/example.js [hunk 1]
```diff
@@ -29,39 +29,45 @@ module.exports = function exampleInlineTagDef(
       var relativePath = unnamedArgs[0];
       var regionName = tagArgs.region || (unnamedArgs.length > 1 ? unnamedArgs[1] : '');
       var title = tagArgs.title || (unnamedArgs.length > 2 ? unnamedArgs[2] : null);
+      var linenums = tagArgs.linenums;
       var stylePattern = tagArgs.stylePattern;  // TODO: not yet implemented here
```
Comment (petebacondarwin) [inline aio/transforms/examples-package/inline-tag-defs/example.js]: We need to implement the "format" attribute stuff here too, which allow the author to add limited classes for styling to the `<code-example>` tag.
This is slightly confused by the fact that in the Jade, the `makeExample` mixin can be chained with a call to `format()`, which (I guess) adds arbitrary attributes to the output node of the mixin. Is that right @wardbell?
As far as I can tell it is only ever called with `format('.')` which indicates `linenums:'false'` (i.e. explicitly no line numbers in the example), and `linenums:XX`, where XX is a number and refers to the start number for this example. Are there any other cases that I am not aware of @wardbell / @Foxandxss ?

Comment (Foxandxss) [inline aio/transforms/examples-package/inline-tag-defs/example.js]: I don't think so Pete. I have only seen the `format('.')` for the reasoning you mentioned.

File: aio/transforms/examples-package/inline-tag-defs/example.js [hunk 2]
```diff
@@ -29,39 +29,45 @@ module.exports = function exampleInlineTagDef(
...
+      if (linenums !== undefined) attributes.push(` linenums=${linenums}`);
 
-      // If still no file then we error
-      if (!exampleFile) {
-        log.error(
-            createDocMessage('Missing example file... relativePath: "' + relativePath + '".', doc));
-        log.error('Example files can be found in: ' + EXAMPLES_FOLDERS.join(', '));
-        return '';
-      }
+      return '<code-example' + attributes.join() + '>\n' + sourceCode + '\n</code-example>';
```
Comment (gkalpak) [inline aio/transforms/examples-package/inline-tag-defs/example.js]: `.join()` --> `.join('')` (by default, arrays are joined with `,`)

**Review by gkalpak — APPROVED** [review]
One minor comment. LGTM otherwise.

File: aio/transforms/examples-package/inline-tag-defs/example.js (hunk 2 above)
Comment (petebacondarwin) [inline aio/transforms/examples-package/inline-tag-defs/example.js]: Oops.

**Comment by petebacondarwin** [comment]
I decided to write some tests for this inline tag and found plenty of other issues. Good old unit tests.

**Comment by petebacondarwin** [comment]
@gkalpak I added some tests and fixed some stuff

**Comment by gkalpak** [comment]
Still LGTM (although the tests are not exhaustive 😛)

**Pull request merged**